Add Model Automation and Dashboarding page (#3751)
ROCm/ROCm: docs/how-to/rocm-for-ai/model-automation-and-dashboarding.rst

.. meta::
   :description: Discover, run, and benchmark deep learning models with AMD MAD -- Model Automation and Dashboarding tool.

>    :keywords: dashboard, machine, deep, container, playground, tune

Reviewer: ```suggestion
   :keywords: dashboard, machine, deep, container, playground, tune, MAD, dashboarding, model automation
```

************************

> Running models using MAD

Reviewer: Is this the right title?
Curious as three models talk about three things you can use MAD for.

Reviewer: Your description in the metadata might be better - not sure

Reviewer: Notice the MAD repo uses the following intro:
MAD (Model Automation and Dashboarding), is a combination of AI/ML model zoo, automation for running the models on various GPU architectures, a mechanism of maintaining historical performance data, and generating dashboards for tracking.

Do we need to closer align? For example, this discusses AI/ML model

************************

The AMD Model Automation and Dashboarding (MAD) tool integrates an AI model zoo with automated execution capabilities
across various GPU architectures. It facilitates performance tracking by maintaining historical performance data and

> generating dashboards for analysis. MAD's source code repository and full documentation can be found at
> `<https://github.com/ROCm/MAD>`__.

Reviewer: ```suggestion
generating dashboards for analysis. MAD's source code repository and complete documentation is at
`<https://github.com/ROCm/MAD>`__.
```

MAD retrieves various models from their repositories and tests their performance in ROCm Docker images. It is an index
of deep learning models optimized for reproducible accuracy and performance on AMD GPUs and accelerators using the ROCm
software stack.

Use MAD to:

> *  Try new models,
> 
> *  Compare performance between patches or architectures, and
> 
> *  Track functionality and performance over time.

Reviewer: ```suggestion
*  Try new models

*  Compare performance between patches or architectures

*  Track functionality and performance over time
```

Getting started with MAD
========================

> Refer to the procedures in :doc:`rocm-install-on-linux:index` to set up your host computer with ROCm. Follow the

Reviewer: The second sentence not necessary as ROCm only runs on Linux.

detailed :doc:`installation instructions <rocm-install-on-linux:install/detailed-install>` for Linux-based platforms.

ROCm Docker images
------------------

> You can find ROCm Docker images for PyTorch and TensorFlow on Docker Hub at
> `rocm/pytorch <https://hub.docker.com/r/rocm/pytorch>`_ and
> `rocm/tensorflow <https://hub.docker.com/r/rocm/tensorflow>`_.

Reviewer: ```suggestion
ROCm Docker images for PyTorch and TensorFlow on Docker Hub are available from
`rocm/pytorch <https://hub.docker.com/r/rocm/pytorch>`_ and
`rocm/tensorflow <https://hub.docker.com/r/rocm/tensorflow>`_.
```

> AMD publishes a unified Docker image at `rocm/vllm <https://hub.docker.com/r/rocm/vllm>`_ that packages
> together vLLM and PyTorch for the AMD Instinct™ MI300X accelerator. This enables users to quickly validate the expected
> inference performance numbers on the MI300X. This Docker image includes:

Reviewer: ```suggestion
AMD publishes a unified Docker image at `rocm/vllm <https://hub.docker.com/r/rocm/vllm>`_ which packages vLLM and PyTorch for the AMD Instinct™ MI300X accelerator. This enables youo quickly validate the expected
inference performance numbers on the MI300X. This Docker image includes:
```

Reviewer: I'm usually careful around phrases like:
AMD publishes...

Perhaps instead:
An AMD unified docker is available... or soemthing

- ROCm

- vLLM

- PyTorch

- Tuning files (CSV format)

See `<https://github.com/ROCm/MAD/tree/develop/benchmark/vllm>`__ for more information.

.. _mad-run-locally:

Using MAD to run models locally
===============================

> The following describes MAD's basic usage and functionalities.

Reviewer: ```suggestion
The following describes MAD basic usage and functionality.
```

   ``models.json`` locally on a Docker host. Refer to
   `MAD/models.json <https://github.com/ROCm/blob/develop/models.json>`_.

   ``run_models.py`` is the main MAD command line interface for running models locally. While the tool has many options,
   running any single model is very easy. To run a model, look for its name or tag in the ``models.json`` and pass it to
   ``run_models.py`` in the form of:

   .. code-block:: shell

      tools/run_models.py [-h] [--tags TAGS] [--timeout TIMEOUT] [--live-output] [--clean-docker-cache] [--keep-alive] [--keep-model-dir] [-o OUTPUT] [--log-level LOG-LEVEL]

   See :ref:`mad-run-args` for the list of options and their descriptions.

For each model in ``models.json``, the script:

* Builds Docker images associated with each model. The images are named
  ``ci-$(model_name)``, and are not removed after the script completes.

* Starts the Docker container, with name, ``container_$(model_name)``.
  The container should automatically be stopped and removed whenever
  the script exits.

* Clones the git ``url`` and runs the ``scripts``.

* Compiles the final ``perf.csv`` and ``perf.html``.

.. _mad-run-args:

Arguments
---------

> The following list of options describe some of MAD capabilities.

Reviewer: Do the options describe MAD capabilities? something awkward

--help, -h

>    Show this help message and exit

Reviewer: strange option
